### PR TITLE
feature/3306 URs on maned dashboard

### PR DESCRIPTION
**Pull request by richard-jones**
# Update requests on maned dashboard

*Please don't delete any sections when completing this PR template; instead enter **N/A** for checkboxes or sections which are not applicable, unless otherwise stated below*

See https://github.com/DOAJ/doajPM/issues/3306

This PR adds update requests to the managing editor's dashboard.  The dashboard can now be filtered by New Application or UR (or display both, which is the default).

## Categorisation

This PR...
- [ ] has scripts to run
- [ ] has migrations to run
- [ ] adds new infrastructure
- [ ] changes the CI pipeline
- [ ] affects the public site
- [x] affects the editorial area
- [ ] affects the publisher area
- [ ] affects the monitoring

## Basic PR Checklist

Instructions for developers:
* For each checklist item, if it is N/A to your PR check the N/A box
* For each item that you have done and confirmed for yourself, check Developer box (including if you have checked the N/A box)

Instructions for reviewers:
* For each checklist item that has been confirmed by the Developer, check the Reviewer box if you agree
* For multiple reviewers, feel free to add your own checkbox with your github username next to it if that helps with review tracking

### Code Style

- No deprecated methods are used
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

- No magic strings/numbers - all strings are in `constants` or `messages` files
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- ES queries are wrapped in a Query object rather than inlined in the code
  - [x] N/A
  - [ ] Developer
  - [x] Reviewer
  
- Where possible our common library functions have been used (e.g. dates manipulated via `dates`)
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Cleaned up commented out code, etc
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

- Urls are constructed with `url_for` not hard-coded
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
### Testing

- Unit tests have been added/modified
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Functional tests have been added/modified
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer
  
- Code has been run manually in development, and functional tests followed locally
  - [ ] N/A
  - [x] Developer
  - [ ] Reviewer

- Have CSS/style changes been implemented?  If they are of a global scope (e.g. on base HTML elements) have the downstream impacts of the change in other areas of the system been considered?
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer

### Documentation

- FeatureMap annotations have been added
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- Documentation updates - if needed - have been identified and prepared for inclusion into main documentation (e.g. added and highlighted/commented as appropriate to this PR)
  - [ ] N/A
  - [x] Developer - updated the dashboard todo list spec https://docs.google.com/spreadsheets/d/1WJLsbtx3wkdNeakrnP3QkvSYls4uFw0lPPzychq7Mo4/edit#gid=0
  - [x] Reviewer
  
- Core model documentation has been added to if needed: https://docs.google.com/spreadsheets/d/1lun2S9vwGbyfy3WjIjgXBm05D-3wWDZ4bp8xiIYfImM/edit
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer

- Events and consumers documentation has been added if needed: https://docs.google.com/spreadsheets/d/1oIeG5vg-blm2MZCE-7YhwulUlSz6TOUeY8jAftdP9JE/edit
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer
  
- The docs for this branch have been generated and pushed to the doc site (see docs/README.md for details)
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

### Release Readiness

- If needed, migration has been created and tested locally
  - [x] N/A
  - [ ] Developer
  - [ ] Reviewer

- Release sheet has been created, and completed as far as is possible https://docs.google.com/spreadsheets/d/1Bqx23J1MwXzjrmAygbqlU3YHxN1Wf7zkkRv14eTVLZQ/edit#gid=581325614
  - [ ] N/A
  - [x] Developer
  - [x] Reviewer

- There has been a recent merge up from `develop` (or other base branch).  List the dates of the merges up from develop below
  - 2024-01-25
  - 2024-03-07

## Testing

1. Review the new functionality: https://doaj.github.io/doaj-docs/feature/3306_urs_on_dashboard/testbook/index.html#dashboard/maned_todo_list

2. Check the new functionality has not impacted non-managing editor dashboards: https://doaj.github.io/doaj-docs/feature/3306_urs_on_dashboard/testbook/index.html#dashboard/editor_todo_list

## Deployment

What deployment considerations are there? (delete any sections you don't need)

### Configuration changes

N/A

### Scripts

N/A

### Migrations

N/A

### Monitoring

N/A

### New Infrastructure

N/A

### Continuous Integration
N/A

### CODE REVIEW

File: portality/bll/services/todo.py [hunk 1]
```diff
@@ -233,6 +242,26 @@ def maned_assign_pending(cls, size, maned_of):
...
         return constants.TODO_MANED_ASSIGN_PENDING, assign_pending, sort_date, 0
 
+    @classmethod
+    def maned_update_requests(cls, size, maned_of):
+        sort_date = "created_date"
+        assign_pending = TodoQuery(
+            musts=[
+                TodoQuery.exists("admin.editor_group"),
+                TodoQuery.cd_older_than(4),
+                # TodoQuery.status([constants.APPLICATION_STATUS_UPDATE_REQUEST]),
```
Comment (RK206) [inline portality/bll/services/todo.py:252]: May need to remove the commented code

File: portality/bll/services/todo.py [hunk 2]
```diff
@@ -233,6 +242,26 @@ def maned_assign_pending(cls, size, maned_of):
...
+                TodoQuery.editor_group(maned_of),
+                TodoQuery.is_update_request()
+            ],
+            must_nots=[
+                TodoQuery.status([constants.APPLICATION_STATUS_ACCEPTED, constants.APPLICATION_STATUS_REJECTED])
+                # TodoQuery.exists("admin.editor")
```
Comment (RK206) [inline portality/bll/services/todo.py:258]: May need to remove commented code

**Review by RK206 — APPROVED** [review]
All looks good to me. Very minor suggestion to remove commented code

**Comment by amdomanska** [comment]
@richard-jones I have tried to recreate this to see the changes. I went to Update Request search interface, chose one of the opened URs, changed the Editor Group and assigned myself as an Editor. After saving the notification "**New application** has been assigned to you" is shown. The new UR is not shown in any of the tabs (All, New Applications, Update Requests"). What am I missing?

image
image
image
image
image

**Pull request closed without merging**